x/evm/keeper/state_transition.go

```diff
@@ -145,19 +145,22 @@ func (k *Keeper) ApplyTransaction(tx *ethtypes.Transaction) (*types.MsgEthereumT
 	res.Hash = txHash.Hex()
 
 	// Set the bloom filter and commit only if transaction is NOT reverted
-	if !res.Reverted {
-		logs := k.GetTxLogs(txHash)
-		res.Logs = types.NewLogsFromEth(logs)
-		// update block bloom filter
-		bloom := k.GetBlockBloomTransient()
-		bloom.Or(bloom, big.NewInt(0).SetBytes(ethtypes.LogsBloom(logs)))
-		k.SetBlockBloomTransient(bloom)
+	logs := k.GetTxLogs(txHash)
+	bloom := k.GetBlockBloomTransient()
+	bloom.Or(bloom, big.NewInt(0).SetBytes(ethtypes.LogsBloom(logs)))
 
+	// Commit and switch to original context
+	if !res.Reverted {
 		commit()
 	}
+	k.ctx = originalCtx
 
 	// refund gas prior to handling the vm error in order to set the updated gas meter
-	k.ctx = originalCtx
+	if !res.Reverted {
+		res.Logs = types.NewLogsFromEth(logs)
+		// update block bloom filter
+		k.SetBlockBloomTransient(bloom)
```

Comment:
This needs to be set in the original ctx otherwise we wont be able to use it in EndBlock

Reply:
I'd suggest adding a comment to the code

Reply:
done